polkit/validate/validate.go

```diff
@@ -0,0 +1,297 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package validate
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io"
+	"strings"
+)
+
+type Element struct {
+	CharData      string     `xml:",chardata"`
+	ExtraAttrs    []xml.Attr `xml:",any,attr"`
+	ExtraChildren []xml.Name `xml:",any"`
+}
+
+type policyConfig struct {
+	XMLName xml.Name
```

Comment:
afaiu from the docs this could be explicity tagged with `xml:"policyconfig"` and the parser would check for that name?

Reply:
I've made this change.  While it does let the xml package reject elements whose base name is not `policyconfig`, it doesn't catch elements named `policyconfig` in a namespace other than the blank namespace.

It's not even clear there is a way to represent the blank namespace using the package's struct tags.

Reply:
On namespaces in general, it looks like they've had a "fix namespaces" tracking bug since 2015: https://github.com/golang/go/issues/13400.